entity-service-client/src/main/java/org/hypertrace/entity/data/service/client/EdsCacheClient.java

```diff
@@ -17,26 +17,35 @@
 import org.hypertrace.entity.data.service.v1.EntityRelationship;
 import org.hypertrace.entity.data.service.v1.EntityRelationships;
 import org.hypertrace.entity.data.service.v1.Query;
+import org.hypertrace.entity.service.client.config.EntityServiceClientCacheConfig;
 import org.hypertrace.entity.service.client.config.EntityServiceClientConfig;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class EdsCacheClient implements EdsClient {
 
   private static final Logger LOG = LoggerFactory.getLogger(EdsCacheClient.class);
-  private final LoadingCache<EdsCacheKey, Optional<EnrichedEntity>> enrichedEntityCache;
-  private final LoadingCache<EdsCacheKey, Optional<Entity>> entityCache;
+  private LoadingCache<EdsCacheKey, Optional<EnrichedEntity>> enrichedEntityCache;
+  private LoadingCache<EdsCacheKey, Optional<Entity>> entityCache;
+  private LoadingCache<EdsTypeAndIdAttributesCacheKey, Optional<String>> entityIdsCache;
 
   private final EntityDataServiceClient client;
 
   EdsCacheClient(EntityServiceClientConfig entityServiceClientConfig) {
     this(new EntityDataServiceClient(entityServiceClientConfig));
+    initCache(entityServiceClientConfig.getCacheConfig());
   }
 
   public EdsCacheClient(EntityDataServiceClient client) {
```

Comment:
When the client application has a EDSClient that it wants to reuse, it should pass the cache configs too. It can pass empty `Config` to use defaults though.

Reply:
See the previous comment.